stdlib/public/Darwin/Foundation/NSDictionary.swift

```diff
@@ -107,19 +107,84 @@ extension Dictionary : _ObjectiveCBridgeable {
     result = builder.take()
   }
 
+  private static func _typePun<T>(index:Int, of buffer: UnsafeBufferPointer<AnyObject>, to: T.Type) -> UnsafeMutablePointer<T> {
+    let typedBase = UnsafeMutableRawPointer(mutating: buffer.baseAddress!).assumingMemoryBound(to: T.self)
```

Comment:
This part is a bit ridiculous, but I'm trying really hard to a) avoid binding the memory at base to T prematurely, and b) avoid doing pointer math myself when Swift can already learn the size and alignment from T itself